src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -16,6 +16,8 @@
 using Microsoft.OpenApi.Services;
 using Microsoft.OpenApi.Writers;
 
+#nullable enable
```

Comment:
instead of relying on directives, could we enable that at the project level to get a consistent experience, and better end to end linting from the compiler?

Reply:
Then I will have to add nullable disable for all files and gradually turn it off

Reply:
yes, I understand this represents much more work.

Reply:
My main goal is to do this gradually, it will also be much easier to review/align with team changes

Reply:
> Then I will have to add nullable disable for all files and gradually turn it off

Let's do this in next PR so it will be easier to do fast review

Reply:
If you want to do it gradually maybe we can setup a branch feature/nullability (from release/2) here, and you can make multiple PRs to it. What I'm worried about is that we end up in a state where nullability is implemented halfway through, and can't publish public previews until the work is fully implemented because of side effects.

Reply:
I understand, but I can’t guarantee it. it will also imply constant conflicts that will have to be resolved, which will further slow down the process.
As far as I know, maui is marked in the same way, gradually. For me, it is better to have partial markings than not to have them at all.

Reply:
Noted. I'm most likely not going to be the one doing the reviews anyway as it's outside of my core responsibilities, and I only followed up because I had originally created the issue.

I'd like @millicentachieng @coseguera and @MaggieKimani1 input on this and how they want to process this large batch of pull requests since they are the one most likely to review and merge your work at this point.